### PR TITLE
config/node: Update status to use metav1.Condition type; Re-add node crd.yaml

### PR DESCRIPTION
**Pull request by swghosh**
1. Change config.NodeStatus.WorkerLatencyProfileStatus.Conditions to use []metav1.Condition 
    - discussion: https://github.com/openshift/cluster-kube-controller-manager-operator/pull/611#discussion_r844502601
2. Re-add config/v1/node-crd.yaml that was reverted
   - as per: https://github.com/openshift/api/pull/1160#issuecomment-1090371608
   - revert was to avoid breaking e2e-upgrades on our CI

Signed-off-by: Swarup Ghosh <swghosh@redhat.com>

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/approve
/label px-approved
/label qe-approved
/label docs-approved
/assign @atiratree

File: config/v1/0000_10_config-operator_01_node.crd.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+              properties:
+                workerLatencyProfileStatus:
+                  description: WorkerLatencyProfileStatus provides the current status of WorkerLatencyProfile
+                  type: object
+                  properties:
+                    conditions:
+                      description: conditions describes the state of the WorkerLatencyProfile and related components (Kubelet or Controller Manager or Kube API Server)
+                      type: array
+                      items:
+                        description: "Condition contains details for one aspect of the current state of this API Resource. --- This struct is intended for direct use as an array at the field path .status.conditions.  For example, type FooStatus struct{     // Represents the observations of a foo's current state.     // Known .status.conditions.type are: \"Available\", \"Progressing\", and \"Degraded\"     // +patchMergeKey=type     // +patchStrategy=merge     // +listType=map     // +listMapKey=type     Conditions []metav1.Condition `json:\"conditions,omitempty\" patchStrategy:\"merge\" patchMergeKey:\"type\" protobuf:\"bytes,1,rep,name=conditions\"` \n     // other fields }"
```
Comment (atiratree) [inline config/v1/0000_10_config-operator_01_node.crd.yaml:65]: needs an update?

Comment (swghosh) [reply config/v1/0000_10_config-operator_01_node.crd.yaml:65]: This `description` is relying on description message from upstream k8s metav1.Condition, whenever I change this description manually running `make generate-with-container` reverts it back to this very message.

Comment (atiratree) [reply config/v1/0000_10_config-operator_01_node.crd.yaml:65]: Ah sorry, I saw  `\"Progressing\", and \"Degraded\"` and thought it was related to our conditions. On a second look this is exact copy from metav1.Condition. So we can keep it as is.

File: config/v1/types_node.go [hunk 2]
```diff
@@ -107,76 +107,36 @@ type WorkerLatencyProfileStatus struct {
...
 	// respective arguments.
-	WorkerLatencyProfileProgressing WorkerLatencyStatusConditionType = "Progressing"
+	MachineConfigProgressing         = "MachineConfigProgressing"
+	KubeControllerManagerProgressing = "KubeControllerManagerProgressing"
+	KubeAPIServerProgressing         = "KubeAPIServerProgressing"
 
 	// Complete indicates whether the component (Kubelet or Controller Manager or Kube API Server)
 	// is successfully updated the respective arguments.
-	WorkerLatencyProfileComplete WorkerLatencyStatusConditionType = "Complete"
+	MachineConfigComplete         = "MachineConfigComplete"
```
Comment (atiratree) [inline config/v1/types_node.go:125]: shouldn't this be a Kubelet?

Comment (swghosh) [reply config/v1/types_node.go:125]: cc: @sairameshv 
Which one would be more appropriate?
`MachineConfigComplete` or `KubeletComplete`? / the alike.

@atiratree the way MCO is actually handling Kubelet args is via Machine Configs. But on the other hand, if we wanna stick to operand naming "kubelet" is more apt.

File: config/v1/0000_10_config-operator_01_node.crd.yaml (hunk 1 above)
Comment (swghosh) [reply config/v1/0000_10_config-operator_01_node.crd.yaml:65]: Alright, thanks!

File: config/v1/types_node.go (hunk 2 above)
Comment (swghosh) [reply config/v1/types_node.go]: @atiratree used "Kubelet..." in 3ffa44f.

Comment (sairameshv) [reply config/v1/types_node.go]: Yeah @swghosh , "kubelet" sounds good.

**Comment by swghosh** [comment]
/remove-hold
as we have both 4.11-ci and 4.11-nightly builds on our CI that's green and includes reverted node crd yaml change.

**Pull request merged**